common/s3util.h

```diff
@@ -245,17 +250,51 @@ class S3Util {
     return read_ratelimit_mb_;
   }
 
+  // For test only.
+  static uint32_t getInstanceCounter() {
+    std::lock_guard<std::mutex> guard(counter_mutex_);
+    return instance_counter_;
+  }
+
  private:
+  struct ThisIsPrivate {
```

Comment:
Why not simply move S3Util() into the private section?

Reply:
The S3Util::BuildS3Util() uses std::make_shared<S3Util>() to create a std::shared_ptr<S3Util>. 
std::make_shared requires the S3Util constructor to be public. So it won't compile if we move constructor to private section.